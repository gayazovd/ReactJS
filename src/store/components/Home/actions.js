export const FETCH_REQUEST_TO_SERVER = 'FETCH_REQUEST_TO_SERVER';

export const fetchDataServer = movies => {
    return {
        type: FETCH_REQUEST_TO_SERVER,
        movies
    }
}