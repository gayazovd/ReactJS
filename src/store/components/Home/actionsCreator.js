import { fetchDataServer } from './actions';
import { network } from '../../../services/network'; 

export const fetchData = url => dispatch => {
        network(url)
            .then(data =>{
                const {data: movies} = data.data;
                dispatch(fetchDataServer(movies))
            }) 
    }
