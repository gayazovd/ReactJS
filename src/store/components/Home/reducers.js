import { FETCH_REQUEST_TO_SERVER } from './actions';

const dataFromServer = (state = [], action) => {
    switch (action.type) {
        case FETCH_REQUEST_TO_SERVER:
            return action.movies;
        default:
            return state;
    }
};

export default dataFromServer;
