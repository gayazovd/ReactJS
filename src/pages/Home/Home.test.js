import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from './index';

configure({ adapter: new Adapter() });

describe('<Home/> component testing',() => {
    const home = shallow(<Home />)
    it('component <Home /> snapshot',() => {
        expect(home).toMatchSnapshot();
    });

    it('component is render', () => {
        expect(home.find("Header")).toHaveLength(1);
        expect(home.find("SearchPanel")).toHaveLength(1);
        expect(home.find("InformationPanel")).toHaveLength(1);
        expect(home.find("MovieList")).toHaveLength(1);
        expect(home.find("Footer")).toHaveLength(1);
    })
})