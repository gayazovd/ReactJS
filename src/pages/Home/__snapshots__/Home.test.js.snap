// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> component testing component <Home /> snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
